lib/galaxy/config/__init__.py

```diff
@@ -225,7 +225,10 @@ def _update_raw_config_from_kwargs(self, kwargs):
 
     def _create_attributes_from_raw_config(self):
         for key, value in self._raw_config.items():
-            setattr(self, key, value)
+            if not hasattr(self, key):
+                setattr(self, key, value)
+            else:
+                log.debug("Can't override existing attribute %s ", key)
```

Comment:
```suggestion
                log.debug("Can't override existing attribute '%s'", key)
```
or should that be a full-blown exception instead?

Reply:
Yeah, probably. I want to think a bit and see if I can come up with a use case for when we want just a warning..  If I can't think of one, I'll change this shortly.

Reply:
It may be safer to raise an Exception now and change the code if/when an use case emerges.

Reply:
Yes, you are right. Changing.

Reply:
Does this look right?
`raise ConfigurationError("Attempting to override existing attribute %s ", key)`

Reply:
I'd write it as:
```python
            if hasattr(self, key):
                raise ConfigurationError("Attempting to override existing attribute '%s'" % key)
            setattr(self, key, value)
```

Reply:
Yes, of course.